compiler/lexer.cpp

```diff
@@ -163,9 +164,43 @@ plungefile(char* name, int try_currentpath, int try_includepaths)
             result = plungequalifiedfile(path);
         }
     }
+
+    set_file_defines(inpfname);
+
     return result;
 }
 
+static void
+set_file_defines(char* file)
+{
+    size_t i, len;
+    char* inptr;
+    char newpath[512], newname[512];
+
+    inptr = NULL;
+    len = strlen(file);
+    for (i = len - 1; i < len; i--) {
+        if (file[i] == '/'
+#if defined WIN32 || defined _WIN32
+            || file[i] == '\\'
+#endif
+        ) {
+            inptr = &file[i + 1];
+            break;
+        }
+    }

```

Comment:
This becomes something like:

```
std::string inptr;
for (size_t i = file.size() - 1; i < file.size(); i--) {
    if (condition) {
        inptr = file.substr(i);
        break;
```

Finally to fix the backslash bug mentioned offline, you could do something like:
```
auto pos = file.find('\\');
while (pos != std::string::npos) {
    file.insert(pos + 1, '\\');
    pos = file.find('\\', pos + 2);
}
```

(Completely untested!)